src/main/java/org/openrewrite/java/testing/mockito/MockitoWhenOnStaticToMockStatic.java

```diff
@@ -0,0 +1,95 @@
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MockitoWhenOnStaticToMockStatic extends Recipe {
+
+    private static final MethodMatcher MOCKITO_WHEN = new MethodMatcher("org.mockito.Mockito when(..)");
+
+    @Override
+    public String getDisplayName() {
+        return "Replace `Mockito.when` on static (non mock) with try-with-resource with MockedStatic";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Replace `Mockito.when` on static (non mock) with try-with-resource with MockedStatic as Mockito4 no longer allows this.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(new UsesType<>("org.mockito.Mockito", true),
+                new JavaIsoVisitor<ExecutionContext>() {
+                    @Override
+                    public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration method, ExecutionContext ctx) {
+                        J.MethodDeclaration m = super.visitMethodDeclaration(method, ctx);
+                        boolean rewrittenWhen = false;
+                        List<Statement> statementsBeforeWhen = new ArrayList<>();
+                        List<Statement> statementsAfterWhen = new ArrayList<>();
+                        for (Statement stmt : m.getBody().getStatements()) {
+                            if (stmt instanceof J.MethodInvocation &&
+                                MOCKITO_WHEN.matches(((J.MethodInvocation) stmt).getSelect())) {
+                                J.MethodInvocation when = (J.MethodInvocation) ((J.MethodInvocation) stmt).getSelect();
+                                if (when.getArguments().get(0) instanceof J.MethodInvocation && ((J.MethodInvocation) when.getArguments().get(0)).getMethodType().getFlags().contains(Flag.Static)) {
+                                    JavaType.FullyQualified arg_fq = TypeUtils.asFullyQualified(when.getArguments().get(0).getType());
+                                    J.Identifier ident = (J.Identifier) ((J.MethodInvocation)when.getArguments().get(0)).getSelect();
+                                    String template = String.format("try(MockedStatic<#{}> mock%s = mockStatic(#{}.class)){\n" +
+                                                                    "    mock%s.when(#{any()}).thenReturn(#{any()});\n" +
+                                                                    "}", arg_fq.getClassName(), arg_fq.getClassName());
+                                    m = JavaTemplate.builder(template)
```

Comment:
I think this `try` is generated at an illegal location, whihc won't help type attribution:
```java
import static org.mockito.Mockito.mockStatic;
import org.mockito.MockedStatic;
import static org.mockito.Mockito.*;
import static org.junit.Assert.assertEquals;
import a.b.A;
import org.openrewrite.java.internal.template.__M__;
import org.openrewrite.java.internal.template.__P__;
class Test{private A aMock;
void test(){
/*__TEMPLATE__*/try(MockedStatic<a.b.A> mockA = mockStatic(a.b.A.class)){
    mockA.when(__P__.<a.b.A>/*__p2__*/p()).thenReturn(__P__.<a.b.A>/*__p3__*/p());
}/*__TEMPLATE_STOP__*/
;}}
```
via `.doBeforeParseTemplate(System.out::println)`.